plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -436,6 +437,9 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
         if (isDeltaLakeTable(table)) {
             throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, format("Cannot query Delta Lake table '%s'", tableName));
         }
+        if (isIcebergTable(table)) {
+            return null;
```

Comment:
Throw "Cannot query Iceberg table" instead of making the system throw "table not found".

Reply:
Unfortunately this doesn't work well with the concept of redirection recently implemented in hive.
If we throw an exception while trying to get the table handle in Hive, the workflow used for in the analyzer for getting the table handle will stop  abruptly before checking for redirections.

Reply:
> If we throw an exception while trying to get the table handle in Hive, the workflow used for in the analyzer for getting the table handle will stop abruptly before checking for redirections.

the redirections should be consulted first.

Reply:
> The current logic first checks whether we're dealing with a materialized view, then with a view and only at the end with a table (that may contain redirections).

An iceberg table is not a materialized view, nor a view, so `get[Materialized]View` should return `Optional.empty` for these calls.

Reply:
At the moment, when checking whether we're dealing with a materialized view or view there is subsequently made a call towards `HiveMetadata#getTableHandle` which would end up in throwing an exception instead of returning `Optional.empty` in case that an exception would be thrown (in a similar fashion as for delta lake tables).

Reply:
Changing the logic of the `StatementAnalyzer` is something I'd avoid to do in this case for retrieving first the table redirections.

For this reason, I have opted to partly replicate the logic of `HiveMetadata#getTableHandle` method in the `PartitionsSystemTableProvider` which offers me the possibility to do a fine grained handling in case of dealing with an iceberg/delta lake table.

One inconvenient of this approach is that we have duplicated logic in `HiveMetadata` and in `PartitionsSystemTableProvider`.

Reply:
> Changing the logic of the `StatementAnalyzer` is something I'd avoid to do in this case for retrieving first the table redirections.

Agreed, this is out of question.